Source/WebCore/css/CSSSelector.h

```diff
@@ -495,20 +521,20 @@ inline void CSSSelector::setValue(const AtomString& value, bool matchLowerCase)
 {
     ASSERT(match() != Tag);
     AtomString matchingValue = matchLowerCase ? value.convertToASCIILowercase() : value;
-    if (!m_hasRareData && matchingValue != value)
+    if (!hasRareData() && matchingValue != value)
         createRareData();
     
     // Need to do ref counting manually for the union.
-    if (!m_hasRareData) {
-        if (m_data.value)
-            m_data.value->deref();
-        m_data.value = value.impl();
-        m_data.value->ref();
+    if (!hasRareData()) {
+        if (valuePointer())
+            valuePointer()->deref();
+        setValuePointer(value.impl());
+        valuePointer()->ref();
         return;
     }
 
-    m_data.rareData->matchingValue = WTFMove(matchingValue);
-    m_data.rareData->serializingValue = value;
+    rareDataPointer()->matchingValue = WTFMove(matchingValue);
+    rareDataPointer()->serializingValue = value;
 }
 
 inline CSSSelector::~CSSSelector()
```

Comment:
With PGO etc some of this stuff could surely go to .cpp (not needed in this patch).